torch/optim/lr_scheduler.py

```diff
@@ -249,26 +251,35 @@ def _reset(self):
         self.cooldown_counter = 0
         self.num_bad_epochs = 0
 
+    """ Status codes to be returned by ReduceLROnPlateau.step() """
+    STATUS_WAITING = 1
+    STATUS_UPDATED_BEST = 2
+    STATUS_REDUCED_LR = 3
+
     def step(self, metrics, epoch=None):
         current = metrics
         if epoch is None:
             epoch = self.last_epoch = self.last_epoch + 1
         self.last_epoch = epoch
 
-        if self.is_better(current, self.best):
-            self.best = current
-            self.num_bad_epochs = 0
-        else:
-            self.num_bad_epochs += 1
-
         if self.in_cooldown:
             self.cooldown_counter -= 1
             self.num_bad_epochs = 0  # ignore any bad epochs in cooldown
+        else:
+            self.num_bad_epochs += 1
 
-        if self.num_bad_epochs > self.patience:
-            self._reduce_lr(epoch)
-            self.cooldown_counter = self.cooldown
+        if self.is_better(current, self.best):
+            self.best = current
             self.num_bad_epochs = 0
+            return self.STATUS_UPDATED_BEST
+        else:
+            if self.num_bad_epochs > self.patience:
+                self._reduce_lr(epoch)
+                self.cooldown_counter = self.cooldown
+                self.num_bad_epochs = 0
+                return self.STATUS_REDUCED_LR
+            else:
+                return self.STATUS_WAITING
```

Comment:
Not a big fan of integer return codes. I'd replace these constants with strings, so they can be used both as an enum, and can be easily printed too